In [1]:
#script for automating the processing of unlabeled images into a labeled dataset

import cv2
import os
import shutil
from pathlib import Path

SOURCE_DIR = Path("../data/unlabeled_crops")
BASE_TARGET = Path("../data/labeled_dataset_final")

MAP = {
    '1': 'a_bikes', '2': 'b_moto', '3': 'c_pass', '4': 'd_light_comm',
    '5': 'e_heavy_rigid', '6': 'f_articulated', '7': 'g_bus', '8': 'h_agri',
    '0': 'discard', ' ': 'discard' # Space or 0 to throw away bad crops
}

In [ ]:
# def start_labeling():
#     for folder in MAP.values():
#         (BASE_TARGET / folder).mkdir(parents=True, exist_ok=True)

#     files = [f for f in os.listdir(SOURCE_DIR) if f.lower().endswith(('.jpg', '.jpeg', '.png'))]
#     print(f"Total images to label: {len(files)}")

#     for i, filename in enumerate(files):
#         img_path = SOURCE_DIR / filename
#         img = cv2.imread(str(img_path))
        
#         display_img = cv2.resize(img, (500, 500)) 
#         cv2.putText(display_img, f"{i}/{len(files)}: {filename}", (10, 30), 
#                     cv2.FONT_HERSHEY_SIMPLEX, 0.7, (0, 255, 0), 2)
        
#         cv2.imshow("GDDKiA Labeler (1-8, Space=Discard, Esc=Exit)", display_img)
        
#         key = cv2.waitKey(0)
#         char = chr(key & 0xFF)

#         if key == 27: # Esc
#             break
#         elif char in MAP:
#             target_path = BASE_TARGET / MAP[char] / filename
#             shutil.move(str(img_path), str(target_path))
#         else:
#             print(f"Invalid key {char}. Use 1-8 or Space.")

#     cv2.destroyAllWindows()

# if __name__ == "__main__":
#     start_labeling()

Total images to label: 0


In [2]:
# print out the count of images in each folder
for folder in MAP.values():
    count = len(list((BASE_TARGET / folder).glob("*.*")))
    print(f"{folder}: {count} images")

a_bikes: 363 images
b_moto: 424 images
c_pass: 11757 images
d_light_comm: 1247 images
e_heavy_rigid: 639 images
f_articulated: 425 images
g_bus: 269 images
h_agri: 67 images
discard: 0 images
discard: 0 images


24.05 
---
    a_bikes: 33 images
    b_moto: 188 images
    c_pass: 11415 images
    d_light_comm: 1099 images
    e_heavy_rigid: 622 images
    f_articulated: 239 images
    g_bus: 194 images
    h_agri: 65 images

---


25.05
---
    a_bikes: 170 images
    b_moto: 370 images
    c_pass: 11535 images
    d_light_comm: 1120 images
    e_heavy_rigid: 616 images
    f_articulated: 250 images
    g_bus: 210 images
    h_agri: 65 images

---

07.06
---
    a_bikes: 363 images
    b_moto: 424 images
    c_pass: 11757 images
    d_light_comm: 1247 images
    e_heavy_rigid: 639 images
    f_articulated: 425 images
    g_bus: 269 images
    h_agri: 67 images